### PR TITLE
Changes meta

### PR DESCRIPTION
**Pull request by AshokDaparthi**
- Added the string and pointer 
- Using "empty" as default to represent NULL for sai_string_t and sai_pointer_t types.

### CODE REVIEW

**Comment by AshokDaparthi** [comment]
@kcudnik - Please review, Please edit if needed.

**Comment by lguohan** [comment]
test this

File: meta/parse.pl [hunk 1]
```diff
@@ -208,6 +208,8 @@ sub ExtractDescription
 sai_int32_t             s32
 sai_uint64_t            u64
 sai_int64_t             s64
+sai_string_t            string
+sai_pointer_t           pointer
```
Comment (kcudnik) [inline meta/parse.pl]: this should be same as in saitypes value so "str" and "ptr"

File: meta/saisanitycheck.c [hunk 2]
```diff
@@ -548,15 +549,14 @@ void check_attr_default_required(
...
-        case SAI_ATTR_VALUE_TYPE_UINT32_LIST:
+             break;
+        case SAI_ATTR_VALUE_TYPE_UINT32_LIST:        
         case SAI_ATTR_VALUE_TYPE_INT32_LIST:
         case SAI_ATTR_VALUE_TYPE_OBJECT_LIST:
         case SAI_ATTR_VALUE_TYPE_ACL_FIELD_DATA_OBJECT_LIST:
         case SAI_ATTR_VALUE_TYPE_ACL_ACTION_DATA_OBJECT_LIST:
         case SAI_ATTR_VALUE_TYPE_ACL_FIELD_DATA_UINT8_LIST:
-
+
```
Comment (kcudnik) [inline meta/saisanitycheck.c]: remove whitespace

File: meta/saisanitycheck.c [hunk 3]
```diff
@@ -719,7 +721,9 @@ void check_attr_default_value_type(
 
             switch (md->attrvaluetype)
             {
-                case SAI_ATTR_VALUE_TYPE_UINT32_LIST:
+                case SAI_ATTR_VALUE_TYPE_STRING:
+                case SAI_ATTR_VALUE_TYPE_POINTER:
```
Comment (kcudnik) [inline meta/saisanitycheck.c]: those are not list types, you can remove them from here

File: meta/saisanitycheck.c [hunk 4]
```diff
@@ -567,6 +567,8 @@ void check_attr_default_required(
             break;
 
         case SAI_ATTR_VALUE_TYPE_UINT8_LIST:
+        case SAI_ATTR_VALUE_TYPE_STRING:
+        case SAI_ATTR_VALUE_TYPE_POINTER:
```
Comment (kcudnik) [inline meta/saisanitycheck.c]: those are not list types, please move them to separate case statement and allow default value type as CONST or NONE only

Comment (AshokDaparthi) [inline meta/saisanitycheck.c]: in case of pointer and string, default value type NONE is used.

File: meta/saisanitycheck.c (hunk 3 above)
Comment (AshokDaparthi) [inline meta/saisanitycheck.c]: Removed

File: meta/saisanitycheck.c (hunk 2 above)
Comment (AshokDaparthi) [inline meta/saisanitycheck.c]: Done

File: meta/parse.pl (hunk 1 above)
Comment (AshokDaparthi) [inline meta/parse.pl]: Modified to str and ptr

**Comment by kcudnik** [comment]
Ack

**Comment by AshokDaparthi** [comment]
We can document saying size not more than size x or we need to define string as below structure 
sai_string_t  { int len;  char *str}.

**Comment by kcudnik** [comment]
We already have s8list struct which could be reused for that purpose which also have abbility to request how long the string is setting initial count to zero, i will ping Guohan to take a look

File: meta/saisanitycheck.c [hunk 5]
```diff
@@ -719,7 +739,7 @@ void check_attr_default_value_type(
 
             switch (md->attrvaluetype)
             {
-                case SAI_ATTR_VALUE_TYPE_UINT32_LIST:
+                case SAI_ATTR_VALUE_TYPE_UINT32_LIST:
```
Comment (lguohan) [inline meta/saisanitycheck.c]: this line is not changed, please remove the trailing whitespaces.

**Comment by lguohan** [comment]
@AshokDaparthi , can we use s8list as suggested by kcudnik? I suggest not including '\0' to get the count for the string, that is how strlen() is doing.

**Comment by AshokDaparthi** [comment]
@lguohan @kcudnik I will remove the sai_string_t and use the sai_s8list_t and update the pull request,

File: inc/saitypes.h [hunk 6]
```diff
@@ -211,6 +211,10 @@ typedef struct _sai_u8_list_t {
     uint8_t *list;
 } sai_u8_list_t;
 
+/**
+ * @brief Defines a s8 list or string.
+ * string should be null terminated and count should include '\0'
```
Comment (kcudnik) [inline inc/saitypes.h]: This seems to be different then common string usage, usually string length don't contain \0 i that would be intuitive behaviour

Comment (AshokDaparthi) [inline inc/saitypes.h]: @lguohan - Based on our discussion we thought of adding the \0 also in count right. Otherwise we can leave it to strlen behavior.

Comment (kcudnik) [inline inc/saitypes.h]: ok

File: meta/saisanitycheck.c [hunk 7]
```diff
@@ -596,15 +612,14 @@ void check_attr_default_required(
         case SAI_ATTR_VALUE_TYPE_UINT64:
         case SAI_ATTR_VALUE_TYPE_MAC:
         case SAI_ATTR_VALUE_TYPE_IP_ADDRESS:
-            break;
-
+             break;
```
Comment (lguohan) [inline meta/saisanitycheck.c]: extra space here?

File: meta/saisanitycheck.c [hunk 8]
```diff
@@ -630,6 +645,9 @@ void check_attr_default_required(
 
             break;
 
+        case SAI_ATTR_VALUE_TYPE_POINTER:
+             break;
```
Comment (lguohan) [inline meta/saisanitycheck.c]: >break; [](start = 13, length = 6)

extra space here?

Comment (AshokDaparthi) [inline meta/saisanitycheck.c]: Removed

File: meta/saisanitycheck.c (hunk 7 above)
Comment (AshokDaparthi) [inline meta/saisanitycheck.c]: Removed

**Pull request merged**